xarray/tests/test_units.py

```diff
@@ -1245,6 +1311,766 @@ def test_dot_dataarray(dtype):
+            converted_y = convert_units(y, units)
+        else:
+            converted_y = y
+
+        if all(compatible_mappings(units, extract_units(variable)).values()):
+            converted_variable = convert_units(variable, units)
+        else:
+            converted_variable = variable
+
+        expected = attach_units(
+            func(strip_units(converted_variable), strip_units(converted_y)), units
+        )
+        actual = func(variable, y)
+
+        assert extract_units(expected) == extract_units(actual)
+        xr.testing.assert_allclose(expected, actual)
+
+    @pytest.mark.parametrize(
+        "unit,error",
+        (
+            pytest.param(1, DimensionalityError, id="no_unit"),
+            pytest.param(
+                unit_registry.dimensionless, DimensionalityError, id="dimensionless"
+            ),
+            pytest.param(unit_registry.s, DimensionalityError, id="incompatible_unit"),
+            pytest.param(
+                unit_registry.cm,
+                None,
+                id="compatible_unit",
+                marks=pytest.mark.xfail(
+                    reason="getitem_with_mask converts to the unit of other"
+                ),
+            ),
+            pytest.param(unit_registry.m, None, id="identical_unit"),
+        ),
+    )
+    @pytest.mark.parametrize(
+        "func", (method("where"), method("_getitem_with_mask")), ids=repr
+    )
+    def test_masking(self, func, unit, error, dtype):
+        base_unit = unit_registry.m
+        array = np.linspace(0, 5, 10).astype(dtype) * base_unit
+        variable = xr.Variable("x", array)
+        cond = np.array([True, False] * 5)
+
+        other = -1 * unit
+
+        if error is not None:
+            with pytest.raises(error):
+                func(variable, cond, other)
+
+            return
+
+        expected = attach_units(
+            func(
+                strip_units(variable),
+                cond,
+                strip_units(
+                    convert_units(
+                        other,
+                        {None: base_unit}
+                        if is_compatible(base_unit, unit)
+                        else {None: None},
+                    )
+                ),
+            ),
+            extract_units(variable),
+        )
+        actual = func(variable, cond, other)
+
+        assert extract_units(expected) == extract_units(actual)
+        xr.testing.assert_identical(expected, actual)
+
+    def test_squeeze(self, dtype):
+        shape = (2, 1, 3, 1, 1, 2)
+        names = list("abcdef")
+        array = np.ones(shape=shape) * unit_registry.m
+        variable = xr.Variable(names, array)
+
+        expected = attach_units(
+            strip_units(variable).squeeze(), extract_units(variable)
+        )
+        actual = variable.squeeze()
+
+        assert extract_units(expected) == extract_units(actual)
+        xr.testing.assert_identical(expected, actual)
+
+        for dim in names:
```

Comment:
this loop doesn't seem to be doing anything.

Reply:
that's true, I wanted to pass `dim` to `squeeze`